joplin/js/CreateContentModal/richTextPlaceholder.js

```diff
@@ -0,0 +1,51 @@
+export default function (){
+  /* - MutationObserver for Wagtail JS handling
+    - This function will handle flagged elements for custom styling and behavior.
+    - Reference: https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver
+      - Includes option to stop observing.
+  */
+  const wagtailFlaggedClassname = 'odd-placeholder'
+  const targetNode = document.getElementById('tab-content');
+
+  const config = { attributes: true, childList: true, subtree: true };
+
+  const callback = function(mutationsList) {
+
+    for (let mutation of mutationsList) {
+      if (mutation.type === 'childList') {
+        let placeholder = mutation.target.querySelector('.public-DraftEditorPlaceholder-inner');
+        if (placeholder) {
+          let parent = placeholder;
+          checkFlaggedAncestors(parent, placeholder);
+        }
+      }
+    }
+
+  }
+
+  function checkFlaggedAncestors(parent, placeholder){
+    // - Here, we check up through all the elements direct parent nodes (ancestors),
+    // - If a placeholder flag is present we add the class acordingly.
+    while (parent = parent.parentNode) {
```

Comment:
I think you can use jQuery here instead of manually traversing the DOM. See `insertWizardData.js`.

Reply:
So, I did look into this and the issue I was having was that I needed to only query on the direct parents, because I didn't want to change the placeholders for all the placeholders with our specified "coa"-class. I wasn't able to get jQuery to specify this, or would have to do another filter operation. 
- This manual approach just checks directly "up" (parent, grandparent... etc), avoiding conflicting "coa"-class elements.

Reply:
You are correct! Thanks for the explanation.